recipes/google-cloud-cpp/2.x/conanfile.py

```diff
@@ -138,23 +138,29 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def requirements(self):
-        self.requires("protobuf/3.21.9", transitive_headers=True)
-        self.requires("grpc/1.50.1", transitive_headers=True)
-        self.requires("nlohmann_json/3.10.0")
-        self.requires("crc32c/1.1.1")
-        self.requires("abseil/20220623.0", transitive_headers=True)
-        self.requires("libcurl/7.88.1")
+        # Updating Protobuf requires updates to the generated code, even within
+        # a minor version.
+        self.requires("protobuf/3.21.12", transitive_headers=True)
+        # Each release of Abseil is a new major version, not guaranteed to be
+        # compatible with previous verssions.
+        self.requires("abseil/[>=20230125.3.0 <20230126]", transitive_headers=True)
+        # The rest are more semver-like and require less pinning.
+        self.requires("grpc/[>=1.50]", transitive_headers=True)
+        self.requires("nlohmann_json/[>=3.10 <4]")
+        self.requires("crc32c/[>=1.1 <2]")
+        self.requires("libcurl/[>=7.47 <9]")
         self.requires("openssl/[>=1.1 <4]")
-        self.requires("zlib/1.2.13")
+        self.requires("zlib/[>=1.2 <2]")
```

Comment:
Conan 2.0.14 will improve the situation here with https://github.com/conan-io/conan/pull/14912 allowing to calculate range intersections of missmatching but compatible ranges, but it's still better if we have as unified set of ranges for libraries as possible - see here for the current ones https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges (and additionaly, zlib's is `[>=1.2.11 <2]`)

Reply:
Done.